File: Data Structures & Algorithms/maximum-product-subarray/submission-3.java
class Solution {
    public int maxProduct(int[] nums) {
        int n = nums.length;
        int ans = Integer.MIN_VALUE;
        for(int i=0;i<n;i++){
            int temp = 1;
            for(int j=i;j<n;j++){
                temp*=nums[j];
                ans = Math.max(ans,temp);
            }
        }
        return ans;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cpp
class Solution {
public:
    int findMaxLength(unordered_set<int> nums, int start){
        int ans = 0;
        while(nums.find(start)!=nums.end()){
            start++;
            ans++;
        }
        return ans;
    }

    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> num_set;
        int ans = 0;
        for(auto& num: nums){
            num_set.insert(num);
        }    

        for(auto& num: num_set){
            if(num_set.find(num-1)!=num_set.end())continue;
            ans = max(ans,findMaxLength(num_set, num));
        }
        return ans;
    }
};
